scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -329,53 +323,38 @@ object Pkg {
 
 @branch trait Ctor extends Tree with Member
 object Ctor {
-  @branch trait Call extends Term
   @ast class Primary(mods: List[Mod],
-                     name: Ctor.Name,
+                     name: Name,
                      paramss: List[List[Term.Param]]) extends Ctor
   @ast class Secondary(mods: List[Mod],
-                       name: Ctor.Name,
+                       name: Name,
                        paramss: List[List[Term.Param]] @nonEmpty,
-                       body: Term) extends Ctor with Stat {
-    checkFields(body.isCtorBody)
-  }
-  @branch trait Ref extends scala.meta.Term.Ref with Ctor.Call
-  val Name = Ref.Name
-  type Name = Ref.Name
-  object Ref {
-    // TODO: current design with Ctor.Name(value) has a problem of sometimes meaningless `value`
-    // for example, q"def this() = ..." is going to have Ctor.Name("this"), because we're parsing
-    // this constructor outside of any enclosure, so we can't call it Ctor.Name("C") or Ctor.Name("D")
-    // an alternative design might reintroduce the Ctor.Ref ast node that would have the structure of:
-    // Ctor.Ref(tpe: Type, ctor: Ctor.Name), where Ctor.Name would be Ctor.Name()
-    // in that design, we also won't have to convert between Type and Ctor.Ref hierarchies, which is a definite plus
-    @ast class Name(value: String @nonEmpty) extends scala.meta.Name with Ref
-    @ast class Select(qual: Term.Ref, name: Name) extends Ref
-    @ast class Project(qual: Type, name: Name) extends Ref
-    @ast class Function(name: Name) extends Ref
-  }
-  def fresh(): Ctor.Name = fresh("fresh")
-  def fresh(prefix: String): Ctor.Name = Ctor.Name(prefix + Fresh.nextId())
+                       init: Init,
+                       stats: List[Stat]) extends Ctor with Stat {
+    checkFields(stats.forall(_.isBlockStat))
+  }
+}
+
+@ast class Init(tpe: Type, name: Name, argss: List[List[Term]]) extends Ref {
```

Comment:
Additionally, it might be worth adding a comment highlighting difference between Init and Term.Apply. Why are both `tpe: Type` and `name: Name` necessary? According to my best understanding, it's necessary for the semantic api so that Ref.symbol references the correct type + ctor symbols, unlike regular Term.Apply